index.bs

```diff
@@ -5080,6 +5157,58 @@ The [=remote end steps=] given |session| and |command parameters| are:
 
 ### Events ### {#module-script-events}
 
+#### The script.message Event #### {#event-script-message}
+
+<dl>
+   <dt>Event Type</dt>
+   <dd>
+      <pre class="cddl local-cddl">
+        script.Message = {
+         method: "script.message",
+         params: script.MessageParameters
+       }
+
+       script.MessageParameters = {
+         channel: script.Channel,
+         data: RemoteValue
+         realm: script.Realm
+       }
+      </pre>
+   </dd>
+</dl>
+
+<div algorithm="remote end event trigger for script.message">
+The [=remote end event trigger=] is the <dfn>emit a script message</dfn> steps,
+given |session|, |realm|, |channel|, and |message|:
+
+1. Let |environment settings| be the [=environment settings object=] whose
+   [=realm execution context=]'s Realm component is |realm|.
+
+1. Let |related browsing contexts| be the result of [=get related browsing
+   contexts=] given |environment settings|.
+
+1. If [=event is enabled=] given |session|, "<code>script.message</code>"
+   and |related browsing contexts|:
+
+  1. Let |data| be the result of [=serialize as a remote value=] given
+     |message|, <code>1</code> as max depth, <code>none</code> as ownership type,
```

Comment:
I wonder if we never want to have ownership here. Given that there might be cases when a client wants to use a returned objects as argument for a successive `callFunction` call a handle would be needed. Right now we allow any kind of value returned via the channel.